src/Sprites/ArtWriter.cpp

```diff
@@ -88,16 +88,23 @@ void ArtFile::WriteAnimation(Stream::SeekableWriter& seekableWriter, const Anima
 
 void ArtFile::WriteFrame(Stream::SeekableWriter& seekableWriter, const Animation::Frame& frame)
 {
-	std::size_t subframeCount = frame.subframes.size();
+	std::size_t layerCount = frame.layers.size();
+
+	//TODO: FIX FACT THAT BITFIELDS ARE NOT BEING WRITTEN CORRECTLY.
```

Comment:
Ahh, do you mean the bit is not set if optional fields are present? Is this information still available here? It might be more than just checking if the optional fields are zero or not (and they might not even be initialized if they were not loaded).

It looks like you're writing raw casted `uint8_t` count fields, rather than using a bitfield struct which would force the count into a 7-bit field.

Reply:
Yeah, anything below this TODO is pretty much wrong right now.